util/types/field_type.go

```diff
@@ -78,6 +78,16 @@ func AggFieldType(tps []*FieldType) *FieldType {
 	return &currType
 }
 
+// AggTypeClass4FieldType aggregates arguments' TypeClass of a multi-argument function, and adjust FieldType's
+// Charset, Collate and Flag if needed.
+func AggTypeClass4FieldType(tps []*FieldType, target *FieldType) TypeClass {
```

Comment:
@spongedu 
built-in func `IF` can be referred,
and 
better address `case-when` first. ^_^